### ISSUE TITLE
iOS Push Notification Deployment Failure on New Projects

### ISSUE THREAD

**Issue by davhad** [FlutterFlow/flutterflow-issues#271]
ios App deployment failure on push notifications activation on new hello world project.

## Current behavior (required)
A brand new project created for testing follows the push notifications instructions and consistently generate the same error.

```
error: Provisioning profile "<Project Name>" doesn't support the Push Notifications capability. (in target 'Runner' from project 'Runner')
    error: Provisioning profile "<Project Name>" doesn't include the aps-environment entitlement. (in target 'Runner' from project 'Runner')
    Result bundle written to path:
/var/folders/w2/rrf5p87d1bbfyphxc7jdnyvh0000gn/T/flutter_tools.UCmUrW/flutter_ios_build_temp_dirAMVHp6/temporary_xcresult_bundle


Error (Xcode): Provisioning profile "<Project Name>" doesn't support the Push Notifications capability.

Error (Xcode): Provisioning profile "<Project Name>" doesn't include the aps-environment entitlement.


It appears that there was a problem signing your application prior to installation on the device.

Verify that the Bundle Identifier in your project is your signing id in Xcode
  open ios/Runner.xcworkspace

Also try selecting 'Product > Build' to fix the problem.
Encountered error while archiving for device.

Build failed :|
Step 8 script 'Flutter build ipa and automatic versioning' exited with status code 1
```

## To Reproduce (required)

1. Create brand new Flutterflow project
2. Firebase added, as well as cloud functions, authentication, etc.
3. App can be previewed, run, and deployment to ios works. 
4. Apple push notifications key created.
5. APN .p8 Key, team ID, and key ID added to Firestore cloud messaging.
6. App ID identifier of type app created with explicit bundle ID: [projectname].ImageNotification (Capital I & N)
7. Enable push notifications toggled on on Flutterflow app.
8. Push notifications cloud functions successfully deployed.
9. App fails with error code below...

### Bug Report Code

ITESiPLqx4VgodxI14fyKMZQhQIgQ0QnU7gnlO1ucx4YC73mOu0uP87BdUNKT/alTW1qJ0yZjjkFzPzohobTIfQECzaVY4Rzk/UwKlCvHAM=

## Screenshots / recordings
![Screenshot 2022-12-23 at 11 30 29](https://user-images.githubusercontent.com/72607/209320458-91fbfd5e-864a-4ffb-96d9-cacdad33fc2e.png)

![Screenshot 2022-12-23 at 11 30 52](https://user-images.githubusercontent.com/72607/209320415-b49667b0-c20b-4e64-a887-769259eb8422.png)

## Your environment

* Version of FlutterFlow used: 1.1.28
* Platform (e.g. Web, MacOS Desktop): MacOS & Web
* Browser name and version: Chrome 108.0.5359.124
* Operating system and version (desktop or mobile): macOS 13.0.1

**Comment by agreaves** [comment]
Hi @davhad, thanks for the bug report! I think you may be missing a step. In the Identifier for the application itself, you need to make sure to select the Push Notifications capability, like so (ignore Sign in With Apple unless you need it):
<img width="1165" alt="image" src="https://user-images.githubusercontent.com/7757399/209387012-f53211e1-25bf-4a49-9fbb-5cdcb5d412e1.png">

That is most likely what is causing the issue. The Push Notifications capability is specified in the App but not in the App Identifier. You can find that menu by going to https://developer.apple.com/account/resources/identifiers/list and then selecting your app's identifier.

**Comment by davhad** [comment]
Thanks Alex and merry Christmas. Fixed now but please note the documentation is very unclear about that. I'm referring to this https://www.youtube.com/watch?v=Vz3aT7dy5Tk:

1. The FF push notifications tutorial video makes no mention of adding the push notifications entitlement on the main app identifier. FF users might fill the gap by adding it to the notification identifier, which wouldn't work.
2. It's unclear if certificates must be added or not. In this case to work they're not needed. But the developer would naturally have a tendency to click on configure and add the certificates.
3. It continues to be unclear why a new push notifications identifier myst be created instead of having  the entitlement added to the main app identifier.

Long story short, the documentation probably needs updating and clarification specially if support will continually insist on referring you to it when there's an issue :)

**Comment by agreaves** [comment]
Hi @davhad, you're absolutely right! It seems our documentation is missing this very important step, so we'll make sure to add it. Thanks again for the report!
